### PR TITLE
Another strategy for onTap() and refresh result in onShow() 

### PR DESCRIPTION
**Pull request by haraldh**
also remove empty functions, so that the super class functions get called.

This fixes the glitch of the match result not shown in the end in the simulator. Might also fix some VivoActive bugs.

### CODE REVIEW

**Comment by matco** [comment]
Nice, thank you!

I hesitated to call the super class constructor a while ago, but it's not done in all Garmin samples so I assumed it is not required.

It's really cool if it fixes the bug with the Vivoactive.

File: source/views/BeginnerView.mc [hunk 1]
```diff
@@ -36,7 +28,8 @@ class BeginnerViewDelegate extends Ui.BehaviorDelegate {
 
 	function manageChoice(player) {
 		match.begin(player);
-		Ui.switchToView(new MatchView(), new MatchViewDelegate(), Ui.SLIDE_IMMEDIATE);
+		var view = new MatchView();
+		Ui.switchToView(view, new MatchViewDelegate(view), Ui.SLIDE_IMMEDIATE);
```
Comment (matco) [inline source/views/BeginnerView.mc]: Is this required? Should all "Delegates" be constructed with their view as first parameter? I don't think so (see all Garmin samples).

If yes, we should initialize all "Delegates" the same way.

**Comment by haraldh** [comment]
You are right.. my fault... should have checked, that you extended the initializer of the delegate with the view. Force pushed another strategy.

**Comment by haraldh** [comment]
Oh, btw, didn't try to compile and test this one... just a quick suggestion.

**Comment by matco** [comment]
I am not sure about what is the right way to initialize a view or a delegate. Your solution -which seems fine- just avoids the problem by removing delegate initialization method. Therefore, I would be more confortable if this is done in a dedicated commit.

Would you mind if I split this PR myself (creation of a commit including only improvements regarding call of superclass methods and removal of useless method at first)?

**Comment by haraldh** [comment]
sure, take what works and what you can use.. I don't collect git commits :)

**Pull request closed without merging**